src/shogun/mathematics/linalg/internal/implementation/Variance.h

```diff
@@ -0,0 +1,236 @@
+	 * \f$\{x_1, ..., x_m\}\f$, this is
+	 * \f$\frac{1}{m-1}\sum_{i=1}^m (x-\bar{x})^2\f$ where
+	 * \f$\bar x=\frac{1}{m}\sum_{i=1}^m x_i\f$
+	 *
+	 * @param a vector of values
+	 * @return variance of given values
+	 */
+	static T compute(SGVector<T> vec)
+	{
+		ASSERT(vec.vlen>1)
+		ASSERT(vec.vector)
+
+		float64_t mean = implementation::mean<Backend::EIGEN3, Vector>::compute(vec);
+
+		float64_t sum_squared_diff=0;
+		for (index_t i=0; i<vec.vlen; ++i)
+			sum_squared_diff+=CMath::pow(vec.vector[i]-mean, 2);
+
+		return sum_squared_diff / (vec.vlen);
+	}
+};
+
+/**
+ * @Brief A generic class that contains two methods for computing the variance of
+ * a matrix.  There's one method for computing column-wise and row-wise variance and one
+ * method for computing element-wise variance
+ */
+template<enum Backend, typename Matrix>
+struct matrix_variance{
+
+	/** Generic scalar type */
+	typedef typename Matrix::Scalar T;
+
+	/** Vector return type */
+	typedef SGVector<T> ReturnTypeVec;
+
+	/** Calculates unbiased empirical variance estimator of given values. Given
+	 * \f$\{x_1, ..., x_m\}\f$, this is
+	 * \f$\frac{1}{m-1}\sum_{i=1}^m (x-\bar{x})^2\f$ where
+	 * \f$\bar x=\frac{1}{m}\sum_{i=1}^m x_i\f$
+	 *
+	 * Computes the variance for each row/col of matrix
+	 *
+	 * @param values vector of values
+	 * @param col_wise if true, every column vector will be used, row vectors
+	 * otherwise
+	 * @return variance of given values
+	 */
+	static ReturnTypeVec compute(Matrix m, bool col_wise);
+
+	/**
+	 * Method that computes the element-wise variance of a matrix
+	 *
+	 * @param m the matrix whose element-wise variance we want to
+	 * compute
+	 * @return the element-wise variance of m
+	 *
+	static T compute(Matrix m);*/
+};
+
+/**
+ * @Brief A specialization of matrix_variance that uses SGMatrix and SGVector as it's types
+ * and uses Eigen3 as it's backend component
+ */
+template<typename Matrix>
+struct matrix_variance<Backend::EIGEN3, Matrix>{
+
+	/** Generic scalar type */
+	typedef typename Matrix::Scalar T;
+
+	/** Vector return type */
+	typedef SGVector<T> ReturnTypeVec;
+
+	/** Calculates unbiased empirical variance estimator of given values. Given
+	 * \f$\{x_1, ..., x_m\}\f$, this is
+	 * \f$\frac{1}{m-1}\sum_{i=1}^m (x-\bar{x})^2\f$ where
+	 * \f$\bar x=\frac{1}{m}\sum_{i=1}^m x_i\f$
+	 *
+	 * Computes the variance for each row/col of matrix
+	 *
+	 * @param values vector of values
+	 * @param col_wise if true, every column vector will be used, row vectors
+	 * otherwise
+	 * @return variance of given values
+	 */
+	static ReturnTypeVec compute(SGMatrix<T> m, bool col_wise){
+
+		ASSERT(m.num_rows>0)
+		ASSERT(m.num_cols>0)
+		ASSERT(m.matrix)
+
+		ReturnTypeVec variance;
+
+		if(col_wise){
+
+			variance = ReturnTypeVec(m.num_cols);
+			SGVector<T> tempVec(m.num_rows);
+
+			for (index_t i=0; i< m.num_cols; ++i){
+
+				//It'd be nice if SGMatrix offered a get_column_vector function that retrurned a SGVector
```

Comment:
please avoid such comments (but just send a patch :) )

BUT as sbove, can you please do this with eigen without writing loops